<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc -->
<title>GenomicIndexUtil (HTSJDK API Documentation)</title>
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="GenomicIndexUtil (HTSJDK API Documentation)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":9,"i1":9};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GenomicIndexUtil.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../htsjdk/samtools/FixBAMFile.html" title="class in htsjdk.samtools"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../htsjdk/samtools/LinearIndex.html" title="class in htsjdk.samtools"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?htsjdk/samtools/GenomicIndexUtil.html" target="_top">Frames</a></li>
<li><a href="GenomicIndexUtil.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">htsjdk.samtools</div>
<h2 title="Class GenomicIndexUtil" class="title">Class GenomicIndexUtil</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>htsjdk.samtools.GenomicIndexUtil</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="typeNameLabel">GenomicIndexUtil</span>
extends java.lang.Object</pre>
<div class="block">Constants and methods used by BAM and Tribble indices</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../htsjdk/samtools/GenomicIndexUtil.html#BIN_GENOMIC_SPAN">BIN_GENOMIC_SPAN</a></span></code>
<div class="block">Reports the total amount of genomic data that any bin can index.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../htsjdk/samtools/GenomicIndexUtil.html#LEVEL_STARTS">LEVEL_STARTS</a></span></code>
<div class="block">What is the starting bin for each level?</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../htsjdk/samtools/GenomicIndexUtil.html#MAX_BINS">MAX_BINS</a></span></code>
<div class="block">Reports the maximum number of bins that can appear in a binning index.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../htsjdk/samtools/GenomicIndexUtil.html#MAX_LINEAR_INDEX_SIZE">MAX_LINEAR_INDEX_SIZE</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../htsjdk/samtools/GenomicIndexUtil.html#UNSET_GENOMIC_LOCATION">UNSET_GENOMIC_LOCATION</a></span></code>
<div class="block">E.g.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../htsjdk/samtools/GenomicIndexUtil.html#GenomicIndexUtil--">GenomicIndexUtil</a></span>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../htsjdk/samtools/GenomicIndexUtil.html#regionToBin-int-int-">regionToBin</a></span>(int&nbsp;beg,
           int&nbsp;end)</code>
<div class="block">calculate the bin given an alignment in [beg,end)
 Described in "The Human Genome Browser at UCSC.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>static java.util.BitSet</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../htsjdk/samtools/GenomicIndexUtil.html#regionToBins-int-int-">regionToBins</a></span>(int&nbsp;startPos,
            int&nbsp;endPos)</code>
<div class="block">Get candidate bins for the specified region</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, equals, finalize, getClass, hashCode, notify, notifyAll, toString, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="BIN_GENOMIC_SPAN">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>BIN_GENOMIC_SPAN</h4>
<pre>public static final&nbsp;int BIN_GENOMIC_SPAN</pre>
<div class="block">Reports the total amount of genomic data that any bin can index.</div>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#htsjdk.samtools.GenomicIndexUtil.BIN_GENOMIC_SPAN">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="LEVEL_STARTS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>LEVEL_STARTS</h4>
<pre>public static final&nbsp;int[] LEVEL_STARTS</pre>
<div class="block">What is the starting bin for each level?</div>
</li>
</ul>
<a name="MAX_BINS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MAX_BINS</h4>
<pre>public static final&nbsp;int MAX_BINS</pre>
<div class="block">Reports the maximum number of bins that can appear in a binning index.</div>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#htsjdk.samtools.GenomicIndexUtil.MAX_BINS">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="MAX_LINEAR_INDEX_SIZE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MAX_LINEAR_INDEX_SIZE</h4>
<pre>public static final&nbsp;int MAX_LINEAR_INDEX_SIZE</pre>
</li>
</ul>
<a name="UNSET_GENOMIC_LOCATION">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>UNSET_GENOMIC_LOCATION</h4>
<pre>public static final&nbsp;int UNSET_GENOMIC_LOCATION</pre>
<div class="block">E.g. for a SAMRecord with no genomic coordinate.</div>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../constant-values.html#htsjdk.samtools.GenomicIndexUtil.UNSET_GENOMIC_LOCATION">Constant Field Values</a></dd>
</dl>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="GenomicIndexUtil--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>GenomicIndexUtil</h4>
<pre>public&nbsp;GenomicIndexUtil()</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="regionToBin-int-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>regionToBin</h4>
<pre>public static&nbsp;int&nbsp;regionToBin(int&nbsp;beg,
                              int&nbsp;end)</pre>
<div class="block">calculate the bin given an alignment in [beg,end)
 Described in "The Human Genome Browser at UCSC. Kent & al. doi: 10.1101/gr.229102 "</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>beg</code> - 0-based start of read (inclusive)</dd>
<dd><code>end</code> - 0-based end of read (exclusive)</dd>
</dl>
</li>
</ul>
<a name="regionToBins-int-int-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>regionToBins</h4>
<pre>public static&nbsp;java.util.BitSet&nbsp;regionToBins(int&nbsp;startPos,
                                            int&nbsp;endPos)</pre>
<div class="block">Get candidate bins for the specified region</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>startPos</code> - 1-based start of target region, inclusive.</dd>
<dd><code>endPos</code> - 1-based end of target region, inclusive.</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>bit set for each bin that may contain SAMRecords in the target region.</dd>
</dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GenomicIndexUtil.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../htsjdk/samtools/FixBAMFile.html" title="class in htsjdk.samtools"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../htsjdk/samtools/LinearIndex.html" title="class in htsjdk.samtools"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?htsjdk/samtools/GenomicIndexUtil.html" target="_top">Frames</a></li>
<li><a href="GenomicIndexUtil.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
