<h3 id="CheckFingerprint">CheckFingerprint</h3>
<section>
<p>Computes a fingerprint from the supplied input file (SAM/BAM or VCF) file and compares it to the expected fingerprint genotypes provided. The key output is a LOD score which represents the relative likelihood of the sequence data originating from the same sample as the genotypes vs. from a random sample.  Two outputs are produced: (1) a summary metrics file that gives metrics at the single sample level (if the input was a VCF) or at the read level (lane or index within a lane) (if the input was a SAM/BAM) versus a set of known genotypes for the expected sample, and (2) a detail metrics file that contains an individual SNP/Haplotype comparison within a fingerprint comparison.  The two files may be specified individually using the SUMMARY_OUTPUT and DETAIL_OUTPUT options.  Alternatively the OUTPUT option may be used instead to give the base of the two output files, with the summary metrics having a file extension 'fingerprinting_summary_metrics' and the detail metrics having a file extension 'fingerprinting_detail_metrics'.</p>
<table>
<tr><th>Option</th><th>Description</th></tr>
<tr><td>INPUT (File)</td><td>Input file SAM/BAM or VCF.  If a VCF is used, it must have at least one sample.  If there are more than one samples in the VCF, the parameter OBSERVED_SAMPLE_ALIAS must be provided in order to indicate which sample's data to use.  If there are no samples in the VCF, an exception will be thrown.  Required. </td></tr>
<tr><td>OBSERVED_SAMPLE_ALIAS (String)</td><td>If the input is a VCF, this parameters used to select which sample's data in the VCF to use.  Default value: null. </td></tr>
<tr><td>OUTPUT (String)</td><td>The base prefix of output files to write.  The summary metrics will have the file extension 'fingerprinting_summary_metrics' and the detail metrics will have the extension 'fingerprinting_detail_metrics'.  Required.  Cannot be used in conjuction with option(s) SUMMARY_OUTPUT (S) DETAIL_OUTPUT (D)</td></tr>
<tr><td>SUMMARY_OUTPUT (File)</td><td>The text file to which to write summary metrics.  Required.  Cannot be used in conjuction with option(s) OUTPUT (O)</td></tr>
<tr><td>DETAIL_OUTPUT (File)</td><td>The text file to which to write detail metrics.  Required.  Cannot be used in conjuction with option(s) OUTPUT (O)</td></tr>
<tr><td>GENOTYPES (File)</td><td>File of genotypes (VCF or GELI) to be used in comparison. May contain any number of genotypes; CheckFingerprint will use only those that are usable for fingerprinting.  Required. </td></tr>
<tr><td>EXPECTED_SAMPLE_ALIAS (String)</td><td>This parameter can be used to specify which sample's genotypes to use from the expected VCF file (the GENOTYPES file).  If it is not supplied, the sample name from the input (VCF or BAM read group header) will be used.  Default value: null. </td></tr>
<tr><td>HAPLOTYPE_MAP (File)</td><td>The file lists a set of SNPs, optionally arranged in high-LD blocks, to be used for fingerprinting. See https://software.broadinstitute.org/gatk/documentation/article?id=9526 for details.  Required. </td></tr>
<tr><td>GENOTYPE_LOD_THRESHOLD (Double)</td><td>When counting haplotypes checked and matching, count only haplotypes where the most likely haplotype achieves at least this LOD.  Default value: 5.0. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>IGNORE_READ_GROUPS (Boolean)</td><td>If the input is a SAM/BAM, and this parameter is true, treat the entire input BAM as one single read group in the calculation, ignoring RG annotations, and producing a single fingerprint metric for the entire BAM.  Default value: false. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
</table>
</section>
