<a id="CollectGcBiasMetrics"/>
<h3>CollectGcBiasMetrics</h3>
<section>
<p>Tool to collect information about GC bias in the reads in a given BAM file. Computes the number of windows (of size specified by WINDOW_SIZE) in the genome at each GC% and counts the number of read starts in each GC bin.  What is output and plotted is the "normalized coverage" in each bin - i.e. the number of reads per window normalized to the average number of reads per window across the whole genome..
<p></p>
<table>
<tr><th>Option</th><th>Description</th></tr>
<tr><td>REFERENCE_SEQUENCE (File)</td><td>The reference sequence fasta file.  Required. </td></tr>
<tr><td>INPUT (File)</td><td>The BAM or SAM file containing aligned reads.  Must be coordinate-sorted.  Required. </td></tr>
<tr><td>OUTPUT (File)</td><td>The text file to write the metrics table to.  Required. </td></tr>
<tr><td>CHART_OUTPUT (File)</td><td>The PDF file to render the chart to.  Required. </td></tr>
<tr><td>SUMMARY_OUTPUT (File)</td><td>The text file to write summary metrics to.  Default value: null. </td></tr>
<tr><td>WINDOW_SIZE (Integer)</td><td>The size of windows on the genome that are used to bin reads.  Default value: 100. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>MINIMUM_GENOME_FRACTION (Double)</td><td>For summary metrics, exclude GC windows that include less than this fraction of the genome.  Default value: 1.0E-5. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>ASSUME_SORTED (Boolean)</td><td>If true, assume that the input file is coordinate sorted, even if the header says otherwise.  Default value: false. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>IS_BISULFITE_SEQUENCED (Boolean)</td><td>Whether the SAM or BAM file consists of bisulfite sequenced reads.    Default value: false. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
</table>
</section>
