<h3 id="CollectRnaSeqMetrics">CollectRnaSeqMetrics</h3>
<section>
<p><b>Produces RNA alignment metrics for a SAM or BAM file.</b>  <p>This tool takes a SAM/BAM file containing the aligned reads from an RNAseq experiment and produces metrics describing the distribution of the bases within the transcripts.  It calculates the total numbers and the percentages of nucleotides within specific genomic regions including untranslated regions (UTRs), introns, intergenic sequences (between discrete genes), and peptide-coding sequences (exons). This tool also determines the numbers of bases that pass quality filters that are specific to Illumina data (PF_BASES); for more information please see the corresponding <a href='https://www.broadinstitute.org/gatk/guide/article?id=6329'>Dictionary</a>GATK entry.</p><p>Other metrics include the median coverage (depth), 5'/3'-biases, and the numbers of reads with the correct/incorrect strand designation. The 5'/3'-bias results from errors introduced by reverse transcriptase enzymes during library construction, ultimately leading to the over-representation of either the 5' or 3' ends of transcripts.  Please see the CollectRnaSeqMetrics <a href='http://broadinstitute.github.io/picard/picard-metric-definitions.html#RnaSeqMetrics'>definitions</a> for details on how these biases are calculated. </p><p>The sequence input must be a valid SAM/BAM file containing RNAseq data aligned by an RNAseq-aware genome aligner such as <a href='http://github.com/alexdobin/STAR'>STAR</a> or <a href='http://ccb.jhu.edu/software/tophat/index.shtml'>TopHat</a>. The tool also requires a REF_FLAT file, a tab-delimited file containing information about the location of RNA transcripts, exon start and stop sites, etc. For more information on the REF_FLAT format, see the following <a href='http://genome.ucsc.edu/goldenPath/gbdDescriptionsOld.html#RefFlat'>description</a>.  Build-specific REF_FLAT files can be obtained <a href='http://hgdownload.cse.ucsc.edu/goldenPath/'>here</a>.</p><pre><h4>Usage example:</h4>java -jar picard.jar CollectRnaSeqMetrics \<br />      I=input.bam \<br />      O=output.RNA_Metrics \<br />      REF_FLAT=ref_flat.txt \<br />      STRAND=SECOND_READ_TRANSCRIPTION_STRAND \<br />      RIBOSOMAL_INTERVALS=ribosomal.interval_list <br /></pre><p>Please see the CollectRnaSeqMetrics <a href='http://broadinstitute.github.io/picard/picard-metric-definitions.html#RnaSeqMetrics'>definitions</a> for a complete description of the metrics produced by this tool.</p><hr /></p>
<table>
<tr><th>Option</th><th>Description</th></tr>
<tr><td>REF_FLAT (File)</td><td>Gene annotations in refFlat form.  Format described here: http://genome.ucsc.edu/goldenPath/gbdDescriptionsOld.html#RefFlat  Required. </td></tr>
<tr><td>RIBOSOMAL_INTERVALS (File)</td><td>Location of rRNA sequences in genome, in interval_list format.  If not specified no bases will be identified as being ribosomal.  Format described <a href="http://samtools.github.io/htsjdk/javadoc/htsjdk/htsjdk/samtools/util/IntervalList.html">here</a>:  Default value: null. </td></tr>
<tr><td>STRAND_SPECIFICITY (StrandSpecificity)</td><td>For strand-specific library prep. For unpaired reads, use FIRST_READ_TRANSCRIPTION_STRAND if the reads are expected to be on the transcription strand.  Required. Possible values: {NONE, FIRST_READ_TRANSCRIPTION_STRAND, SECOND_READ_TRANSCRIPTION_STRAND} </td></tr>
<tr><td>MINIMUM_LENGTH (Integer)</td><td>When calculating coverage based values (e.g. CV of coverage) only use transcripts of this length or greater.  Default value: 500. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>CHART_OUTPUT (File)</td><td>The PDF file to write out a plot of normalized position vs. coverage.  Default value: null. </td></tr>
<tr><td>IGNORE_SEQUENCE (String)</td><td>If a read maps to a sequence specified with this option, all the bases in the read are counted as ignored bases.  These reads are not counted as   Default value: null. This option may be specified 0 or more times. </td></tr>
<tr><td>RRNA_FRAGMENT_PERCENTAGE (Double)</td><td>This percentage of the length of a fragment must overlap one of the ribosomal intervals for a read or read pair to be considered rRNA.  Default value: 0.8. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>METRIC_ACCUMULATION_LEVEL (MetricAccumulationLevel)</td><td>The level(s) at which to accumulate metrics.    Default value: [ALL_READS]. This option can be set to 'null' to clear the default value. Possible values: {ALL_READS, SAMPLE, LIBRARY, READ_GROUP} This option may be specified 0 or more times. This option can be set to 'null' to clear the default list. </td></tr>
<tr><td>INPUT (File)</td><td>Input SAM or BAM file.  Required. </td></tr>
<tr><td>OUTPUT (File)</td><td>File to write the output to.  Required. </td></tr>
<tr><td>ASSUME_SORTED (Boolean)</td><td>If true (default), then the sort order in the header file will be ignored.  Default value: true. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>STOP_AFTER (Long)</td><td>Stop after processing N reads, mainly for debugging.  Default value: 0. This option can be set to 'null' to clear the default value. </td></tr>
</table>
</section>
