<h3 id="CollectIlluminaBasecallingMetrics">CollectIlluminaBasecallingMetrics</h3>
<section>
<p>Collects Illumina Basecalling metrics for a sequencing run.  <p>This tool will produce per-barcode and per-lane basecall metrics for each sequencing run.  Mean values for each metric are determined using data from all of the tiles.  This tool requires the following data, LANE(#), BASECALLS_DIR, READ_STRUCTURE, and an input file listing the sample barcodes.  Program will provide metrics including: the total numbers of bases, reads, and clusters, as well as the fractions of each bases, reads, and clusters that passed Illumina quality filters (PF) both per barcode and per lane.  For additional information on Illumina's PF quality metric, please see the corresponding <a href='https://www.broadinstitute.org/gatk/guide/article?id=6329'>GATK Dictionary entry</a>.</p> <p>The input barcode_list.txt file is a file containing all of the sample and molecular barcodes and can be obtained from the <a href='http://broadinstitute.github.io/picard/command-line-overview.html#ExtractIlluminaBarcodes'>ExtractIlluminaBarcodes</a> tool.  </p>Note: Metrics labeled as percentages are actually expressed as fractions!  <h4>Usage example:</h4><pre>java -jar picard.jar CollectIlluminaBasecallingMetrics \<br />      BASECALLS_DIR=/BaseCalls/ \<br />      LANE=001 \<br />      READ_STRUCTURE=25T8B25T \<br />      INPUT=barcode_list.txt </pre><p>Please see the CollectIlluminaBasecallingMetrics <a href='http://broadinstitute.github.io/picard/picard-metric-definitions.html#IlluminaBasecallingMetrics'>definitions</a> for a complete description of the metrics produced by this tool.  </p><hr /></p>
<table>
<tr><th>Option</th><th>Description</th></tr>
<tr><td>BASECALLS_DIR (File)</td><td>The Illumina basecalls output directory from which data are read  Required. </td></tr>
<tr><td>BARCODES_DIR (File)</td><td>The barcodes directory with _barcode.txt files (generated by ExtractIlluminaBarcodes). If not set, use BASECALLS_DIR.   Default value: null. </td></tr>
<tr><td>LANE (Integer)</td><td>The lane whose data will be read  Required. </td></tr>
<tr><td>INPUT (File)</td><td>The file containing barcodes to expect from the run - barcodeData.#  Default value: null. </td></tr>
<tr><td>READ_STRUCTURE (String)</td><td>A description of the logical structure of clusters in an Illumina Run, i.e. a description of the structure IlluminaBasecallsToSam assumes the  data to be in. It should consist of integer/character pairs describing the number of cycles and the type of those cycles (B for Sample Barcode, M for molecular barcode, T for Template, and S for skip).  E.g. If the input data consists of 80 base clusters and we provide a read structure of "28T8M8B8S28T" then the sequence may be split up into four reads:
* read one with 28 cycles (bases) of template
* read two with 8 cycles (bases) of molecular barcode (ex. unique molecular barcode)
* read three with 8 cycles (bases) of sample barcode
* 8 cycles (bases) skipped.
* read four with 28 cycles (bases) of template
The skipped cycles would NOT be included in an output SAM/BAM file or in read groups therein.  Required. </td></tr>
<tr><td>OUTPUT (File)</td><td>The file to which the collected metrics are written  Default value: null. </td></tr>
</table>
</section>
