<h3 id="CollectRrbsMetrics">CollectRrbsMetrics</h3>
<section>
<p><b>Collects metrics from reduced representation bisulfite sequencing (Rrbs) data.</b>  <p>This tool uses reduced representation bisulfite sequencing (Rrbs) data to determine cytosine methylation status across all reads of a genomic DNA sequence.  For a primer on bisulfite sequencing and cytosine methylation, please see the corresponding <a href='https://www.broadinstitute.org/gatk/guide/article?id=6330'>GATK Dictionary entry</a>. </p><p>Briefly, bisulfite reduction converts un-methylated cytosine (C) to uracil (U) bases.  Methylated sites are not converted because they are resistant to bisulfite reduction.  Subsequent to PCR amplification of the reaction products, bisulfite reduction manifests as [C -> T (+ strand) or G -> A (- strand)] base conversions.  Thus, conversion rates can be calculated from the reads as follows: [CR = converted/(converted + unconverted)]. Since methylated cytosines are protected against Rrbs-mediated conversion, the methylation rate (MR) is as follows:[MR = unconverted/(converted + unconverted) = (1 - CR)].</p><p>The CpG CollectRrbsMetrics tool outputs three files including summary and detail metrics tables as well as a PDF file containing four graphs. These graphs are derived from the summary table and include a comparison of conversion rates for both CpG and non-CpG sites, the distribution of total numbers of CpG sites as a function of the CpG conversion rates, the distribution of CpG sites by the level of read coverage (depth), and the numbers of reads discarded resulting from either exceeding the mismatch rate or size (too short).  The detailed metrics table includes the coordinates of all of the CpG sites for the experiment as well as the conversion rates observed for each site.</p><h4>Usage example:</h4><pre>java -jar picard.jar CollectRrbsMetrics \<br />      R=reference_sequence.fasta \<br />      I=input.bam \<br />      M=basename_for_metrics_files</pre><p>Please see the CollectRrbsMetrics <a href='https://broadinstitute.github.io/picard/picard-metric-definitions.html#RrbsCpgDetailMetrics'>definitions</a> for a complete description of both the detail and summary metrics produced by this tool.</p><hr /></p>
<table>
<tr><th>Option</th><th>Description</th></tr>
<tr><td>INPUT (File)</td><td>The BAM or SAM file containing aligned reads. Must be coordinate sorted  Required. </td></tr>
<tr><td>METRICS_FILE_PREFIX (String)</td><td>Base name for output files  Required. </td></tr>
<tr><td>REFERENCE (File)</td><td>The reference sequence fasta file  Required. </td></tr>
<tr><td>MINIMUM_READ_LENGTH (Integer)</td><td>Minimum read length  Default value: 5. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>C_QUALITY_THRESHOLD (Integer)</td><td>Threshold for base quality of a C base before it is considered  Default value: 20. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>NEXT_BASE_QUALITY_THRESHOLD (Integer)</td><td>Threshold for quality of a base next to a C before the C base is considered  Default value: 10. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>MAX_MISMATCH_RATE (Double)</td><td>Maximum percentage of mismatches in a read for it to be considered, with a range of 0-1  Default value: 0.1. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>SEQUENCE_NAMES (String)</td><td>Set of sequence names to consider, if not specified all sequences will be used  Default value: null. This option may be specified 0 or more times. </td></tr>
<tr><td>ASSUME_SORTED (Boolean)</td><td>If true, assume that the input file is coordinate sorted even if the header says otherwise.  Default value: false. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>METRIC_ACCUMULATION_LEVEL (MetricAccumulationLevel)</td><td>The level(s) at which to accumulate metrics.    Default value: [ALL_READS]. This option can be set to 'null' to clear the default value. Possible values: {ALL_READS, SAMPLE, LIBRARY, READ_GROUP} This option may be specified 0 or more times. This option can be set to 'null' to clear the default list. </td></tr>
</table>
</section>
