<h3 id="CollectHsMetrics">CollectHsMetrics</h3>
<section>
<p><b>Collects hybrid-selection (HS) metrics for a SAM or BAM file.</b>  This tool takes a SAM/BAM file input and collects metrics that are specific for sequence datasets generated through hybrid-selection. Hybrid-selection (HS) is the most commonly used technique to capture exon-specific sequences for targeted sequencing experiments such as exome sequencing; for more information, please see the corresponding <a href='http://www.broadinstitute.org/gatk/guide/article?id=6331'>GATK Dictionary entry</a>. </p> <p>This tool requires an aligned SAM or BAM file as well as bait and target interval files in Picard interval_list format. You should use the bait and interval files that correspond to the capture kit that was used to generate the capture libraries for sequencing, which can generally be obtained from the manufacturer of the kit. If the baits and target intervals are provided in BED format, you can convert them to the Picard interval_list format using Picard's <a href='http://broadinstitute.github.io/picard/command-line-overview.html#BedToIntervalList'>BedToInterval</a> tool. </p><p>If a reference sequence is provided, this program will calculate both AT_DROPOUT and GC_DROPOUT metrics. Dropout metrics are an attempt to measure the reduced representation of reads, in regions that deviate from 50% G/C content. This reduction in the number of aligned reads is due to the increased numbers of errors associated with sequencing regions with excessive or deficient numbers of G/C bases, ultimately leading to poor mapping efficiencies and lowcoverage in the affected regions. </p><p>If you are interested in getting G/C content and mean sequence depth information for every target interval, use the PER_TARGET_COVERAGE option. </p><p>Note that coverage measurements are capped at ~32K to constrain memory usage.</p><h4>Usage Example:</h4><pre>java -jar picard.jar CollectHsMetrics \<br />      I=input.bam \<br />      O=hs_metrics.txt \<br />      R=reference_sequence.fasta \<br />      BAIT_INTERVALS=bait.interval_list \<br />      TARGET_INTERVALS=target.interval_list</pre> <p>Please see the CollectHsMetrics <a href='http://broadinstitute.github.io/picard/picard-metric-definitions.html#HsMetrics'>definitions</a> for a complete description of the metrics produced by this tool.</p> <hr /></p>
<table>
<tr><th>Option</th><th>Description</th></tr>
<tr><td>BAIT_INTERVALS (File)</td><td>An interval list file that contains the locations of the baits used.  Default value: null. This option must be specified at least 1 times. </td></tr>
<tr><td>BAIT_SET_NAME (String)</td><td>Bait set name. If not provided it is inferred from the filename of the bait intervals.  Default value: null. </td></tr>
<tr><td>MINIMUM_MAPPING_QUALITY (Integer)</td><td>Minimum mapping quality for a read to contribute coverage.  Default value: 20. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>MINIMUM_BASE_QUALITY (Integer)</td><td>Minimum base quality for a base to contribute coverage.  Default value: 20. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>CLIP_OVERLAPPING_READS (Boolean)</td><td>True if we are to clip overlapping reads, false otherwise.  Default value: true. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>TARGET_INTERVALS (File)</td><td>An interval list file that contains the locations of the targets.  Default value: null. This option must be specified at least 1 times. </td></tr>
<tr><td>INPUT (File)</td><td>An aligned SAM or BAM file.  Required. </td></tr>
<tr><td>OUTPUT (File)</td><td>The output file to write the metrics to.  Required. </td></tr>
<tr><td>METRIC_ACCUMULATION_LEVEL (MetricAccumulationLevel)</td><td>The level(s) at which to accumulate metrics.  Default value: [ALL_READS]. This option can be set to 'null' to clear the default value. Possible values: {ALL_READS, SAMPLE, LIBRARY, READ_GROUP} This option may be specified 0 or more times. This option can be set to 'null' to clear the default list. </td></tr>
<tr><td>PER_TARGET_COVERAGE (File)</td><td>An optional file to output per target coverage information to.  Default value: null. </td></tr>
<tr><td>PER_BASE_COVERAGE (File)</td><td>An optional file to output per base coverage information to. The per-base file contains one line per target base and can grow very large. It is not recommended for use with large target sets.  Default value: null. </td></tr>
<tr><td>NEAR_DISTANCE (Integer)</td><td>The maximum distance between a read and the nearest probe/bait/amplicon for the read to be considered 'near probe' and included in percent selected.  Default value: 250. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>COVERAGE_CAP (Integer)</td><td>Parameter to set a max coverage limit for Theoretical Sensitivity calculations. Default is 200.  Default value: 200. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>SAMPLE_SIZE (Integer)</td><td>Sample Size used for Theoretical Het Sensitivity sampling. Default is 10000.  Default value: 10000. This option can be set to 'null' to clear the default value. </td></tr>
</table>
</section>
