<h3 id="CrosscheckFingerprints">CrosscheckFingerprints</h3>
<section>
<p>Checks if all fingerprints within a set of files appear to come from the same individual.The fingerprints are calculated initially at the readgroup level (if present) but can be"rolled-up" by library, sample or file, to increase power and provide results at the desired resolution. Regular output is in a "Moltenized" format, one row per comparison. In this format the output will include the LOD score and also tumor-aware LOD score which can help assess identity even in the presence of a severe LOH sample with high purity.A matrix output is also availalble to facilitate visual inspection of crosscheck results.
A separate CLP, ClusterCrosscheckMetrics, can cluster the results as a connected graphaccording to LOD greater than a threshold. </p>
<table>
<tr><th>Option</th><th>Description</th></tr>
<tr><td>INPUT (File)</td><td>One or more input files (or lists of files) to compare fingerprints for.  Default value: null. This option may be specified 0 or more times. </td></tr>
<tr><td>OUTPUT (File)</td><td>Optional output file to write metrics to. Default is to write to stdout.  Default value: null. </td></tr>
<tr><td>MATRIX_OUTPUT (File)</td><td>Optional output file to write matrix of LOD scores to. This is less informative than the metrics output and only contains Normal-Normal LOD score (i.e. doesn't account for Loss of heterogeneity).It is however sometimes easier to use visually.  Default value: null. </td></tr>
<tr><td>HAPLOTYPE_MAP (File)</td><td>The file lists a set of SNPs, optionally arranged in high-LD blocks, to be used for fingerprinting. See https://software.broadinstitute.org/gatk/documentation/article?id=9526 for details.  Required. </td></tr>
<tr><td>LOD_THRESHOLD (Double)</td><td>If any two groups (with the same sample name) match with a LOD score lower than the threshold the program will exit with a non-zero code to indicate error. Program will also exit with an error if it finds two groups with different sample name that match with a LOD score greater than -LOD_THRESHOLD.

LOD score 0 means equal likelihoodthat the groups match vs. come from different individuals, negative LOD scores mean N logs more likely that the groups are from different individuals, and positive numbers mean N logs more likely that the groups are from the sample individual.   Default value: 0.0. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>CROSSCHECK_BY (DataType)</td><td>Specificies which data-type should be used as the basic comparison unit. Fingerprints from readgroups can be "rolled-up" to the LIBRARY, SAMPLE, or FILE level before being compared. Fingerprints from VCF can be be compared by SAMPLE or FILE.  Default value: READGROUP. This option can be set to 'null' to clear the default value. Possible values: {FILE, SAMPLE, LIBRARY, READGROUP} </td></tr>
<tr><td>NUM_THREADS (Integer)</td><td>The number of threads to use to process files and generate Fingerprints.  Default value: 1. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>ALLOW_DUPLICATE_READS (Boolean)</td><td>Allow the use of duplicate reads in performing the comparison. Can be useful when duplicate marking has been overly aggressive and coverage is low.  Default value: false. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>GENOTYPING_ERROR_RATE (Double)</td><td>Assumed genotyping error rate that provides a floor on the probability that a genotype comes from the expected sample.  Default value: 0.01. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>OUTPUT_ERRORS_ONLY (Boolean)</td><td>If true then only groups that do not relate to each other as expected will have their LODs reported.  Default value: false. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>LOSS_OF_HET_RATE (Double)</td><td>The rate at which a heterozygous genotype in a normal sample turns into a homozygous (via loss of heterozygosity) in the tumor (model assumes independent events, so this needs to be larger than reality).  Default value: 0.5. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>EXPECT_ALL_GROUPS_TO_MATCH (Boolean)</td><td>Expect all groups' fingerprints to match, irrespective of their sample names.  By default (with this value set to false), groups (readgroups, libraries, files, or samples) with different sample names are expected to mismatch, and those with the same sample name are expected to match.   Default value: false. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>EXIT_CODE_WHEN_MISMATCH (Integer)</td><td>When one or more mismatches between groups is detected, exit with this value instead of 0.  Default value: 1. This option can be set to 'null' to clear the default value. </td></tr>
</table>
</section>
