<h3 id="EstimateLibraryComplexity">EstimateLibraryComplexity</h3>
<section>
<p>Estimates the numbers of unique molecules in a sequencing library.  <p>This tool outputs quality metrics for a sequencing library preparation.Library complexity refers to the number of unique DNA fragments present in a given library.  Reductions in complexity resulting from PCR amplification during library preparation will ultimately compromise downstream analyses via an elevation in the number of duplicate reads.  PCR-associated duplication artifacts can result from: inadequate amounts of starting material (genomic DNA, cDNA, etc.), losses during cleanups, and size selection issues.  Duplicate reads can also arise from optical duplicates resulting from sequencing-machine optical sensor artifacts.</p>  <p>This tool attempts to estimate library complexity from sequence of read pairs alone.  Reads are sorted by the first N bases (5 by default) of the first read and then the first N bases of the second read of a pair.   Read pairs are considered to be duplicates if they match each other with no gaps and an overall mismatch rate less than or equal to MAX_DIFF_RATE (0.03 by default).  Reads of poor quality are filtered out to provide a more accurate estimate.  The filtering removes reads with any poor quality bases as defined by a read's MIN_MEAN_QUALITY (20 is the default value) across either the first or second read.  Unpaired reads are ignored in this computation.</p> <p>The algorithm attempts to detect optical duplicates separately from PCR duplicates and excludes these in the calculation of library size.  Also, since there is no alignment information used in this algorithm, an additional filter is applied to the data as follows.  After examining all reads, a histogram is built in which the number of reads in a duplicate set is compared with the number of of duplicate sets.   All bins that contain exactly one duplicate set are then removed from the histogram as outliers prior to the library size estimation.  </p><h4>Usage example:</h4><pre>java -jar picard.jar EstimateLibraryComplexity \<br />     I=input.bam \<br />     O=est_lib_complex_metrics.txt</pre>Please see the documentation for the companion <a href='https://broadinstitute.github.io/picard/command-line-overview.html#MarkDuplicates'>MarkDuplicates</a> tool.<hr /></p>
<table>
<tr><th>Option</th><th>Description</th></tr>
<tr><td>INPUT (File)</td><td>One or more files to combine and estimate library complexity from. Reads can be mapped or unmapped.  Default value: null. This option may be specified 0 or more times. </td></tr>
<tr><td>OUTPUT (File)</td><td>Output file to writes per-library metrics to.  Required. </td></tr>
<tr><td>MIN_IDENTICAL_BASES (Integer)</td><td>The minimum number of bases at the starts of reads that must be identical for reads to be grouped together for duplicate detection.  In effect total_reads / 4^max_id_bases reads will be compared at a time, so lower numbers will produce more accurate results but consume exponentially more memory and CPU.  Default value: 5. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>MAX_DIFF_RATE (Double)</td><td>The maximum rate of differences between two reads to call them identical.  Default value: 0.03. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>MIN_MEAN_QUALITY (Integer)</td><td>The minimum mean quality of the bases in a read pair for the read to be analyzed. Reads with lower average quality are filtered out and not considered in any calculations.  Default value: 20. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>MAX_GROUP_RATIO (Integer)</td><td>Do not process self-similar groups that are this many times over the mean expected group size. I.e. if the input contains 10m read pairs and MIN_IDENTICAL_BASES is set to 5, then the mean expected group size would be approximately 10 reads.  Default value: 500. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>BARCODE_TAG (String)</td><td>Barcode SAM tag (ex. BC for 10X Genomics)  Default value: null. </td></tr>
<tr><td>READ_ONE_BARCODE_TAG (String)</td><td>Read one barcode SAM tag (ex. BX for 10X Genomics)  Default value: null. </td></tr>
<tr><td>READ_TWO_BARCODE_TAG (String)</td><td>Read two barcode SAM tag (ex. BX for 10X Genomics)  Default value: null. </td></tr>
<tr><td>MAX_READ_LENGTH (Integer)</td><td>The maximum number of bases to consider when comparing reads (0 means no maximum).  Default value: 0. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>MIN_GROUP_COUNT (Integer)</td><td>Minimum number group count.  On a per-library basis, we count the number of groups of duplicates that have a particular size.  Omit from consideration any count that is less than this value.  For example, if we see only one group of duplicates with size 500, we omit it from the metric calculations if MIN_GROUP_COUNT is set to two.  Setting this to two may help remove technical artifacts from the library size calculation, for example, adapter dimers.  Default value: 2. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>READ_NAME_REGEX (String)</td><td>Regular expression that can be used to parse read names in the incoming SAM file. Read names are parsed to extract three variables: tile/region, x coordinate and y coordinate. These values are used to estimate the rate of optical duplication in order to give a more accurate estimated library size. Set this option to null to disable optical duplicate detection. The regular expression should contain three capture groups for the three variables, in order. It must match the entire read name. Note that if the default regex is specified, a regex match is not actually done, but instead the read name  is split on colon character. For 5 element names, the 3rd, 4th and 5th elements are assumed to be tile, x and y values. For 7 element names (CASAVA 1.8), the 5th, 6th, and 7th elements are assumed to be tile, x and y values.  Default value: <optimized capture of last three ':' separated fields as numeric values>. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>OPTICAL_DUPLICATE_PIXEL_DISTANCE (Integer)</td><td>The maximum offset between two duplicate clusters in order to consider them optical duplicates. The default is appropriate for unpatterned versions of the Illumina platform. For the patterned flowcell models, 2500 is moreappropriate. For other platforms and models, users should experiment to find what works best.  Default value: 100. This option can be set to 'null' to clear the default value. </td></tr>
</table>
</section>
