<h3 id="IlluminaBasecallsToSam">IlluminaBasecallsToSam</h3>
<section>
<p>Transforms raw Illumina sequencing data into an unmapped SAM or BAM file.<p>The IlluminaBaseCallsToSam program collects, demultiplexes, and sorts reads across all of the tiles of a lane via barcode to produce an unmapped SAM/BAM file.  An unmapped BAM file is often referred to as a uBAM.  All barcode, sample, and library data is provided in the LIBRARY_PARAMS file.  Note, this LIBRARY_PARAMS file should be formatted according to the specifications indicated below.  The following is an example of a properly formmated LIBRARY_PARAMS file:</p>BARCODE_1	OUTPUT	SAMPLE_ALIAS	LIBRARY_NAME
AAAAAAAA	SA_AAAAAAAA.bam	SA_AAAAAAAA	LN_AAAAAAAA
AAAAGAAG	SA_AAAAGAAG.bam	SA_AAAAGAAG	LN_AAAAGAAG
AACAATGG	SA_AACAATGG.bam	SA_AACAATGG	LN_AACAATGG
N	SA_non_indexed.bam	SA_non_indexed	LN_NNNNNNNN
 <p>The BARCODES_DIR file is produced by the <a href='http://broadinstitute.github.io/picard/command-line-overview.html#ExtractIlluminaBarcodes'>ExtractIlluminaBarcodes</a> tool for each lane of a flow cell.</p>  <h4>Usage example:</h4><pre>java -jar picard.jar IlluminaBasecallsToSam \<br />      BASECALLS_DIR=/BaseCalls/ \<br />      LANE=001 \<br />      READ_STRUCTURE=25T8B25T \<br />      RUN_BARCODE=run15 \<br />      IGNORE_UNEXPECTED_BARCODES=true \<br />      LIBRARY_PARAMS=library.params </pre><hr /></p>
<table>
<tr><th>Option</th><th>Description</th></tr>
<tr><td>BASECALLS_DIR (File)</td><td>The basecalls directory.   Required. </td></tr>
<tr><td>BARCODES_DIR (File)</td><td>The barcodes directory with _barcode.txt files (generated by ExtractIlluminaBarcodes). If not set, use BASECALLS_DIR.   Default value: null. </td></tr>
<tr><td>LANE (Integer)</td><td>Lane number.   Required. </td></tr>
<tr><td>OUTPUT (File)</td><td>Deprecated (use LIBRARY_PARAMS).  The output SAM or BAM file. Format is determined by extension.  Required.  Cannot be used in conjuction with option(s) LIBRARY_PARAMS BARCODE_PARAMS</td></tr>
<tr><td>RUN_BARCODE (String)</td><td>The barcode of the run.  Prefixed to read names.  Required. </td></tr>
<tr><td>SAMPLE_ALIAS (String)</td><td>Deprecated (use LIBRARY_PARAMS).  The name of the sequenced sample  Required.  Cannot be used in conjuction with option(s) LIBRARY_PARAMS BARCODE_PARAMS</td></tr>
<tr><td>READ_GROUP_ID (String)</td><td>ID used to link RG header record with RG tag in SAM record.  If these are unique in SAM files that get merged, merge performance is better.  If not specified, READ_GROUP_ID will be set to <first 5 chars of RUN_BARCODE>.<LANE> .  Default value: null. </td></tr>
<tr><td>LIBRARY_NAME (String)</td><td>Deprecated (use LIBRARY_PARAMS).  The name of the sequenced library  Default value: null.  Cannot be used in conjuction with option(s) LIBRARY_PARAMS BARCODE_PARAMS</td></tr>
<tr><td>SEQUENCING_CENTER (String)</td><td>The name of the sequencing center that produced the reads.  Used to set the RG.CN tag.  Default value: BI. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>RUN_START_DATE (Date)</td><td>The start date of the run.  Default value: null. </td></tr>
<tr><td>PLATFORM (String)</td><td>The name of the sequencing technology that produced the read.  Default value: illumina. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>READ_STRUCTURE (String)</td><td>A description of the logical structure of clusters in an Illumina Run, i.e. a description of the structure IlluminaBasecallsToSam assumes the  data to be in. It should consist of integer/character pairs describing the number of cycles and the type of those cycles (B for Sample Barcode, M for molecular barcode, T for Template, and S for skip).  E.g. If the input data consists of 80 base clusters and we provide a read structure of "28T8M8B8S28T" then the sequence may be split up into four reads:
* read one with 28 cycles (bases) of template
* read two with 8 cycles (bases) of molecular barcode (ex. unique molecular barcode)
* read three with 8 cycles (bases) of sample barcode
* 8 cycles (bases) skipped.
* read four with 28 cycles (bases) of template
The skipped cycles would NOT be included in an output SAM/BAM file or in read groups therein.  Required. </td></tr>
<tr><td>BARCODE_PARAMS (File)</td><td>Deprecated (use LIBRARY_PARAMS).  Tab-separated file for creating all output BAMs for barcoded run with single IlluminaBasecallsToSam invocation.  Columns are BARCODE, OUTPUT, SAMPLE_ALIAS, and LIBRARY_NAME.  Row with BARCODE=N is used to specify a file for no barcode match  Required.  Cannot be used in conjuction with option(s) LIBRARY_PARAMS SAMPLE_ALIAS (ALIAS) OUTPUT (O) LIBRARY_NAME (LIB)</td></tr>
<tr><td>LIBRARY_PARAMS (File)</td><td>Tab-separated file for creating all output BAMs for a lane with single IlluminaBasecallsToSam invocation.  The columns are OUTPUT, SAMPLE_ALIAS, and LIBRARY_NAME, BARCODE_1, BARCODE_2 ... BARCODE_X where X = number of barcodes per cluster (optional).  Row with BARCODE_1 set to 'N' is used to specify a file for no barcode match.  You may also provide any 2 letter RG header attributes (excluding PU, CN, PL, and DT)  as columns in this file and the values for those columns will be inserted into the RG tag for the BAM file created for a given row.  Required.  Cannot be used in conjuction with option(s) SAMPLE_ALIAS (ALIAS) OUTPUT (O) LIBRARY_NAME (LIB) BARCODE_PARAMS</td></tr>
<tr><td>ADAPTERS_TO_CHECK (IlluminaAdapterPair)</td><td>Which adapters to look for in the read.  Default value: [INDEXED, DUAL_INDEXED, NEXTERA_V2, FLUIDIGM]. This option can be set to 'null' to clear the default value. Possible values: {PAIRED_END, INDEXED, SINGLE_END, NEXTERA_V1, NEXTERA_V2, DUAL_INDEXED, FLUIDIGM, TRUSEQ_SMALLRNA, ALTERNATIVE_SINGLE_END} This option may be specified 0 or more times. This option can be set to 'null' to clear the default list. </td></tr>
<tr><td>FIVE_PRIME_ADAPTER (String)</td><td>For specifying adapters other than standard Illumina  Default value: null. </td></tr>
<tr><td>THREE_PRIME_ADAPTER (String)</td><td>For specifying adapters other than standard Illumina  Default value: null. </td></tr>
<tr><td>NUM_PROCESSORS (Integer)</td><td>The number of threads to run in parallel. If NUM_PROCESSORS = 0, number of cores is automatically set to the number of cores available on the machine. If NUM_PROCESSORS < 0, then the number of cores used will be the number available on the machine less NUM_PROCESSORS.  Default value: 0. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>FIRST_TILE (Integer)</td><td>If set, this is the first tile to be processed (used for debugging).  Note that tiles are not processed in numerical order.  Default value: null. </td></tr>
<tr><td>TILE_LIMIT (Integer)</td><td>If set, process no more than this many tiles (used for debugging).  Default value: null. </td></tr>
<tr><td>FORCE_GC (Boolean)</td><td>If true, call System.gc() periodically.  This is useful in cases in which the -Xmx value passed is larger than the available memory.  Default value: true. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>APPLY_EAMSS_FILTER (Boolean)</td><td>Apply EAMSS filtering to identify inappropriately quality scored bases towards the ends of reads and convert their quality scores to Q2.  Default value: true. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>MAX_READS_IN_RAM_PER_TILE (Integer)</td><td>Configure SortingCollections to store this many records before spilling to disk. For an indexed run, each SortingCollection gets this value/number of indices.  Default value: 1200000. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>MINIMUM_QUALITY (Integer)</td><td>The minimum quality (after transforming 0s to 1s) expected from reads.  If qualities are lower than this value, an error is thrown.The default of 2 is what the Illumina's spec describes as the minimum, but in practice the value has been observed lower.  Default value: 2. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>INCLUDE_NON_PF_READS (Boolean)</td><td>Whether to include non-PF reads  Default value: true. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>IGNORE_UNEXPECTED_BARCODES (Boolean)</td><td>Whether to ignore reads whose barcodes are not found in LIBRARY_PARAMS.  Useful when outputting BAMs for only a subset of the barcodes in a lane.  Default value: false. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>MOLECULAR_INDEX_TAG (String)</td><td>The tag to use to store any molecular indexes.  If more than one molecular index is found, they will be concatenated and stored here.  Default value: RX. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>MOLECULAR_INDEX_BASE_QUALITY_TAG (String)</td><td>The tag to use to store any molecular index base qualities.  If more than one molecular index is found, their qualities will be concatenated and stored here (.i.e. the number of "M" operators in the READ_STRUCTURE)  Default value: QX. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>TAG_PER_MOLECULAR_INDEX (String)</td><td>The list of tags to store each molecular index.  The number of tags should match the number of molecular indexes.  Default value: null. This option may be specified 0 or more times. </td></tr>
</table>
</section>
