<h3 id="RevertSam">RevertSam</h3>
<section>
<p>Reverts SAM or BAM files to a previous state.  This tool removes or restores certain properties of the SAM records, including alignment information, which can be used to produce an unmapped BAM (uBAM) from a previously aligned BAM. It is also capable of restoring the original quality scores of a BAM file that has already undergone base quality score recalibration (BQSR) if theoriginal qualities were retained.<h4>Example with single output:</h4><pre>java -jar picard.jar RevertSam \<br />     I=input.bam \<br />     O=reverted.bam</pre>Output format is BAM by default, or SAM or CRAM if the input path ends with '.sam' or '.cram', respectively.<h4>Example outputting by read group with output map:</h4><pre>java -jar picard.jar RevertSam \<br />     I=input.bam \<br />     OUTPUT_BY_READGROUP=true \<br />     OUTPUT_MAP=reverted_bam_paths.tsv</pre>Will output a BAM/SAM file per read group. By default, all outputs will be in BAM format. However, a SAM file will be produced instead for any read group mapped in OUTPUT_MAP to a path ending with '.sam'. A CRAM file will be produced for any read group mapped to a path ending with '.cram'. <h4>Example outputting by read group without output map:</h4><pre>java -jar picard.jar RevertSam \<br />     I=input.bam \<br />     OUTPUT_BY_READGROUP=true \<br />     O=/write/reverted/read/group/bams/in/this/dir</pre>Will output a BAM/SAM file per read group. By default, all outputs will be in BAM format. However, outputs will be in SAM format if the input path ends with '.sam', or CRAM format if it ends with '.cram'. This behaviour can be overriden with OUTPUT_BY_READGROUP_FILE_FORMAT option.<hr /></p>
<table>
<tr><th>Option</th><th>Description</th></tr>
<tr><td>INPUT (File)</td><td>The input SAM/BAM file to revert the state of.  Required. </td></tr>
<tr><td>OUTPUT (File)</td><td>The output SAM/BAM file to create, or an output directory if OUTPUT_BY_READGROUP is true.  Required.  Cannot be used in conjuction with option(s) OUTPUT_MAP (OM)</td></tr>
<tr><td>OUTPUT_MAP (File)</td><td>Tab separated file with two columns, READ_GROUP_ID and OUTPUT, providing file mapping only used if OUTPUT_BY_READGROUP is true.  Required.  Cannot be used in conjuction with option(s) OUTPUT (O)</td></tr>
<tr><td>OUTPUT_BY_READGROUP (Boolean)</td><td>When true, outputs each read group in a separate file.  Default value: false. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>OUTPUT_BY_READGROUP_FILE_FORMAT (FileType)</td><td>When using OUTPUT_BY_READGROUP, the output file format can be set to a certain format.  Default value: dynamic. This option can be set to 'null' to clear the default value. Possible values: {sam, bam, cram, dynamic} </td></tr>
<tr><td>SORT_ORDER (SortOrder)</td><td>The sort order to create the reverted output file with.  Default value: queryname. This option can be set to 'null' to clear the default value. Possible values: {unsorted, queryname, coordinate, duplicate, unknown} </td></tr>
<tr><td>RESTORE_ORIGINAL_QUALITIES (Boolean)</td><td>True to restore original qualities from the OQ field to the QUAL field if available.  Default value: true. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>REMOVE_DUPLICATE_INFORMATION (Boolean)</td><td>Remove duplicate read flags from all reads.  Note that if this is true and REMOVE_ALIGNMENT_INFORMATION==false,  the output may have the unusual but sometimes desirable trait of having unmapped reads that are marked as duplicates.  Default value: true. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>REMOVE_ALIGNMENT_INFORMATION (Boolean)</td><td>Remove all alignment information from the file.  Default value: true. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>ATTRIBUTE_TO_CLEAR (String)</td><td>When removing alignment information, the set of optional tags to remove.  Default value: [NM, UQ, PG, MD, MQ, SA, MC, AS]. This option can be set to 'null' to clear the default value. This option may be specified 0 or more times. This option can be set to 'null' to clear the default list. </td></tr>
<tr><td>SANITIZE (Boolean)</td><td>WARNING: This option is potentially destructive. If enabled will discard reads in order to produce a consistent output BAM. Reads discarded include (but are not limited to) paired reads with missing mates, duplicated records, records with mismatches in length of bases and qualities. This option can only be enabled if the output sort order is queryname and will always cause sorting to occur.  Default value: false. This option can be set to 'null' to clear the default value. Possible values: {true, false} </td></tr>
<tr><td>MAX_DISCARD_FRACTION (Double)</td><td>If SANITIZE=true and higher than MAX_DISCARD_FRACTION reads are discarded due to sanitization thenthe program will exit with an Exception instead of exiting cleanly. Output BAM will still be valid.  Default value: 0.01. This option can be set to 'null' to clear the default value. </td></tr>
<tr><td>SAMPLE_ALIAS (String)</td><td>The sample alias to use in the reverted output file.  This will override the existing sample alias in the file and is used only if all the read groups in the input file have the same sample alias   Default value: null. </td></tr>
<tr><td>LIBRARY_NAME (String)</td><td>The library name to use in the reverted output file.  This will override the existing sample alias in the file and is used only if all the read groups in the input file have the same library name   Default value: null. </td></tr>
</table>
</section>
