<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc -->
<title>GenotypeLikelihoods (HTSJDK API Documentation)</title>
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="GenotypeLikelihoods (HTSJDK API Documentation)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":9,"i1":9,"i2":10,"i3":41,"i4":9,"i5":9,"i6":9,"i7":9,"i8":41,"i9":9,"i10":10,"i11":10,"i12":10,"i13":10,"i14":9,"i15":10,"i16":42,"i17":10,"i18":9,"i19":9,"i20":9,"i21":10};
var tabs = {65535:["t0","All Methods"],1:["t1","Static Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"],32:["t6","Deprecated Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GenotypeLikelihoods.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-all.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../htsjdk/variant/variantcontext/GenotypeBuilder.html" title="class in htsjdk.variant.variantcontext"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.GenotypeLikelihoodsAllelePair.html" title="class in htsjdk.variant.variantcontext"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?htsjdk/variant/variantcontext/GenotypeLikelihoods.html" target="_top">Frames</a></li>
<li><a href="GenotypeLikelihoods.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">htsjdk.variant.variantcontext</div>
<h2 title="Class GenotypeLikelihoods" class="title">Class GenotypeLikelihoods</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>htsjdk.variant.variantcontext.GenotypeLikelihoods</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="typeNameLabel">GenotypeLikelihoods</span>
extends java.lang.Object</pre>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.GenotypeLikelihoodsAllelePair.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods.GenotypeLikelihoodsAllelePair</a></span></code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected static java.util.Map&lt;java.lang.Integer,java.util.List&lt;java.util.List&lt;java.lang.Integer&gt;&gt;&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#anyploidPloidyToPLIndexToAlleleIndices">anyploidPloidyToPLIndexToAlleleIndices</a></span></code>
<div class="block">A cache of PL index to a list of alleles for any ploidy.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#MAX_DIPLOID_ALT_ALLELES_THAT_CAN_BE_GENOTYPED">MAX_DIPLOID_ALT_ALLELES_THAT_CAN_BE_GENOTYPED</a></span></code>
<div class="block">The maximum number of diploid alternate alleles that we can represent as genotype likelihoods</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#MAX_PL">MAX_PL</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected static int[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#PLindexConversion">PLindexConversion</a></span></code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t1" class="tableTab"><span><a href="javascript:show(1);">Static Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t6" class="tableTab"><span><a href="javascript:show(32);">Deprecated Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>protected static java.util.List&lt;java.util.List&lt;java.lang.Integer&gt;&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#calculateAnyploidPLcache-int-int-">calculateAnyploidPLcache</a></span>(int&nbsp;altAlleles,
                        int&nbsp;ploidy)</code>
<div class="block">Calculate the cache of allele indices for each PL index for a ploidy.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#calculatePLindex-int-int-">calculatePLindex</a></span>(int&nbsp;allele1Index,
                int&nbsp;allele2Index)</code>&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#equals-java.lang.Object-">equals</a></span>(java.lang.Object&nbsp;aThat)</code>&nbsp;</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>static <a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#fromGLField-java.lang.String-">fromGLField</a></span>(java.lang.String&nbsp;GLs)</code>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>static <a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#fromLog10Likelihoods-double:A-">fromLog10Likelihoods</a></span>(double[]&nbsp;log10Likelihoods)</code>&nbsp;</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>static <a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#fromPLField-java.lang.String-">fromPLField</a></span>(java.lang.String&nbsp;PLs)</code>&nbsp;</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>static <a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#fromPLs-int:A-">fromPLs</a></span>(int[]&nbsp;pls)</code>&nbsp;</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>static <a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.GenotypeLikelihoodsAllelePair.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods.GenotypeLikelihoodsAllelePair</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getAllelePair-int-">getAllelePair</a></span>(int&nbsp;PLindex)</code>
<div class="block">Get the diploid allele index pair for the given PL index</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code>static <a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.GenotypeLikelihoodsAllelePair.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods.GenotypeLikelihoodsAllelePair</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getAllelePairUsingDeprecatedOrdering-int-">getAllelePairUsingDeprecatedOrdering</a></span>(int&nbsp;PLindex)</code>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code>static java.util.List&lt;java.lang.Integer&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getAlleles-int-int-">getAlleles</a></span>(int&nbsp;PLindex,
          int&nbsp;ploidy)</code>
<div class="block">Get the allele ploidy indices for the given PL index
 Must use the same ploidy as @see #initializeAnyploidPLIndexToAlleleIndices</div>
</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code>java.util.EnumMap&lt;<a href="../../../htsjdk/variant/variantcontext/GenotypeType.html" title="enum in htsjdk.variant.variantcontext">GenotypeType</a>,java.lang.Double&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getAsMap-boolean-">getAsMap</a></span>(boolean&nbsp;normalizeFromLog10)</code>&nbsp;</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code>int[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getAsPLs--">getAsPLs</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getAsString--">getAsString</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i13" class="rowColor">
<td class="colFirst"><code>double[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getAsVector--">getAsVector</a></span>()</code>
<div class="block">The genotypes likelihoods in -10log10(x) vector format.</div>
</td>
</tr>
<tr id="i14" class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getGQLog10FromLikelihoods-int-double:A-">getGQLog10FromLikelihoods</a></span>(int&nbsp;iOfChoosenGenotype,
                         double[]&nbsp;likelihoods)</code>&nbsp;</td>
</tr>
<tr id="i15" class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getLog10GQ-htsjdk.variant.variantcontext.Genotype-java.util.List-">getLog10GQ</a></span>(<a href="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</a>&nbsp;genotype,
          java.util.List&lt;<a href="../../../htsjdk/variant/variantcontext/Allele.html" title="class in htsjdk.variant.variantcontext">Allele</a>&gt;&nbsp;vcAlleles)</code>&nbsp;</td>
</tr>
<tr id="i16" class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getLog10GQ-htsjdk.variant.variantcontext.GenotypeType-">getLog10GQ</a></span>(<a href="../../../htsjdk/variant/variantcontext/GenotypeType.html" title="enum in htsjdk.variant.variantcontext">GenotypeType</a>&nbsp;genotype)</code>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
</td>
</tr>
<tr id="i17" class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getLog10GQ-htsjdk.variant.variantcontext.Genotype-htsjdk.variant.variantcontext.VariantContext-">getLog10GQ</a></span>(<a href="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</a>&nbsp;genotype,
          <a href="../../../htsjdk/variant/variantcontext/VariantContext.html" title="class in htsjdk.variant.variantcontext">VariantContext</a>&nbsp;context)</code>&nbsp;</td>
</tr>
<tr id="i18" class="altColor">
<td class="colFirst"><code>static int[]</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#getPLIndecesOfAlleles-int-int-">getPLIndecesOfAlleles</a></span>(int&nbsp;allele1Index,
                     int&nbsp;allele2Index)</code>
<div class="block">get the PL indexes (AA, AB, BB) for the given allele pair; assumes allele1Index &lt;= allele2Index.</div>
</td>
</tr>
<tr id="i19" class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#initializeAnyploidPLIndexToAlleleIndices-int-int-">initializeAnyploidPLIndexToAlleleIndices</a></span>(int&nbsp;altAlleles,
                                        int&nbsp;ploidy)</code>
<div class="block">Initialize cache of allele anyploid indices
 If initialized multiple times with the same ploidy, the alternate alleles from the last initialization will be used</div>
</td>
</tr>
<tr id="i20" class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#numLikelihoods-int-int-">numLikelihoods</a></span>(int&nbsp;numAlleles,
              int&nbsp;ploidy)</code>
<div class="block">Compute how many likelihood elements are associated with the given number of alleles
 Equivalent to asking in how many ways N non-negative integers can add up to P is S(N,P)
 where P = ploidy (number of chromosomes) and N = total # of alleles.</div>
</td>
</tr>
<tr id="i21" class="rowColor">
<td class="colFirst"><code>java.lang.String</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html#toString--">toString</a></span>()</code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, finalize, getClass, hashCode, notify, notifyAll, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="MAX_PL">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MAX_PL</h4>
<pre>public static final&nbsp;int MAX_PL</pre>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../constant-values.html#htsjdk.variant.variantcontext.GenotypeLikelihoods.MAX_PL">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="MAX_DIPLOID_ALT_ALLELES_THAT_CAN_BE_GENOTYPED">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>MAX_DIPLOID_ALT_ALLELES_THAT_CAN_BE_GENOTYPED</h4>
<pre>public static final&nbsp;int MAX_DIPLOID_ALT_ALLELES_THAT_CAN_BE_GENOTYPED</pre>
<div class="block">The maximum number of diploid alternate alleles that we can represent as genotype likelihoods</div>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../constant-values.html#htsjdk.variant.variantcontext.GenotypeLikelihoods.MAX_DIPLOID_ALT_ALLELES_THAT_CAN_BE_GENOTYPED">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a name="anyploidPloidyToPLIndexToAlleleIndices">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>anyploidPloidyToPLIndexToAlleleIndices</h4>
<pre>protected static final&nbsp;java.util.Map&lt;java.lang.Integer,java.util.List&lt;java.util.List&lt;java.lang.Integer&gt;&gt;&gt; anyploidPloidyToPLIndexToAlleleIndices</pre>
<div class="block">A cache of PL index to a list of alleles for any ploidy.
 For example, for a ploidy of 3, the allele lists for each PL index is:
 {0,0,0}, {0,0,1}, {0,1,1}, {1,1,1}, {0,0,2}, {0,1,2}, {1,1,2}, {0,2,2}, {1,2,2}, {2,2,2}</div>
</li>
</ul>
<a name="PLindexConversion">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>PLindexConversion</h4>
<pre>protected static final&nbsp;int[] PLindexConversion</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="fromPLField-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromPLField</h4>
<pre>public static final&nbsp;<a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</a>&nbsp;fromPLField(java.lang.String&nbsp;PLs)</pre>
</li>
</ul>
<a name="fromGLField-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromGLField</h4>
<pre>@Deprecated
public static final&nbsp;<a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</a>&nbsp;fromGLField(java.lang.String&nbsp;GLs)</pre>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
</li>
</ul>
<a name="fromLog10Likelihoods-double:A-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromLog10Likelihoods</h4>
<pre>public static final&nbsp;<a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</a>&nbsp;fromLog10Likelihoods(double[]&nbsp;log10Likelihoods)</pre>
</li>
</ul>
<a name="fromPLs-int:A-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>fromPLs</h4>
<pre>public static final&nbsp;<a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods</a>&nbsp;fromPLs(int[]&nbsp;pls)</pre>
</li>
</ul>
<a name="getAsVector--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAsVector</h4>
<pre>public&nbsp;double[]&nbsp;getAsVector()</pre>
<div class="block">The genotypes likelihoods in -10log10(x) vector format.  pr{AA} = x, this
 vector returns math.log10(x) for each of the genotypes.  Can return null if the
 genotype likelihoods are "missing".</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>genotypes likelihoods in negative log10 vector format</dd>
</dl>
</li>
</ul>
<a name="getAsPLs--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAsPLs</h4>
<pre>public&nbsp;int[]&nbsp;getAsPLs()</pre>
</li>
</ul>
<a name="toString--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre>public&nbsp;java.lang.String&nbsp;toString()</pre>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code>toString</code>&nbsp;in class&nbsp;<code>java.lang.Object</code></dd>
</dl>
</li>
</ul>
<a name="getAsString--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAsString</h4>
<pre>public&nbsp;java.lang.String&nbsp;getAsString()</pre>
</li>
</ul>
<a name="equals-java.lang.Object-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>equals</h4>
<pre>public&nbsp;boolean&nbsp;equals(java.lang.Object&nbsp;aThat)</pre>
<dl>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code>equals</code>&nbsp;in class&nbsp;<code>java.lang.Object</code></dd>
</dl>
</li>
</ul>
<a name="getAsMap-boolean-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAsMap</h4>
<pre>public&nbsp;java.util.EnumMap&lt;<a href="../../../htsjdk/variant/variantcontext/GenotypeType.html" title="enum in htsjdk.variant.variantcontext">GenotypeType</a>,java.lang.Double&gt;&nbsp;getAsMap(boolean&nbsp;normalizeFromLog10)</pre>
</li>
</ul>
<a name="getLog10GQ-htsjdk.variant.variantcontext.GenotypeType-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLog10GQ</h4>
<pre>@Deprecated
public&nbsp;double&nbsp;getLog10GQ(<a href="../../../htsjdk/variant/variantcontext/GenotypeType.html" title="enum in htsjdk.variant.variantcontext">GenotypeType</a>&nbsp;genotype)</pre>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
<div class="block">This is really dangerous and returns completely wrong results for genotypes from a multi-allelic context.
 Use <code>getLog10GQ(Genotype,VariantContext)</code>
  or <code>getLog10GQ(Genotype,List&lt;Allele&gt;)</code> in place of it.

 If you <strong>know</strong> you're biallelic, use <code>getGQLog10FromLikelihoods</code> directly.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>genotype</code> - - actually a genotype type (no call, hom ref, het, hom var)</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>an unsafe quantity that could be negative. In the bi-allelic case, the GQ resulting from best minus next best (if the type is the best).</dd>
</dl>
</li>
</ul>
<a name="getLog10GQ-htsjdk.variant.variantcontext.Genotype-java.util.List-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLog10GQ</h4>
<pre>public&nbsp;double&nbsp;getLog10GQ(<a href="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</a>&nbsp;genotype,
                         java.util.List&lt;<a href="../../../htsjdk/variant/variantcontext/Allele.html" title="class in htsjdk.variant.variantcontext">Allele</a>&gt;&nbsp;vcAlleles)</pre>
</li>
</ul>
<a name="getLog10GQ-htsjdk.variant.variantcontext.Genotype-htsjdk.variant.variantcontext.VariantContext-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLog10GQ</h4>
<pre>public&nbsp;double&nbsp;getLog10GQ(<a href="../../../htsjdk/variant/variantcontext/Genotype.html" title="class in htsjdk.variant.variantcontext">Genotype</a>&nbsp;genotype,
                         <a href="../../../htsjdk/variant/variantcontext/VariantContext.html" title="class in htsjdk.variant.variantcontext">VariantContext</a>&nbsp;context)</pre>
</li>
</ul>
<a name="getGQLog10FromLikelihoods-int-double:A-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGQLog10FromLikelihoods</h4>
<pre>public static&nbsp;double&nbsp;getGQLog10FromLikelihoods(int&nbsp;iOfChoosenGenotype,
                                               double[]&nbsp;likelihoods)</pre>
</li>
</ul>
<a name="calculateAnyploidPLcache-int-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>calculateAnyploidPLcache</h4>
<pre>protected static&nbsp;java.util.List&lt;java.util.List&lt;java.lang.Integer&gt;&gt;&nbsp;calculateAnyploidPLcache(int&nbsp;altAlleles,
                                                                                            int&nbsp;ploidy)</pre>
<div class="block">Calculate the cache of allele indices for each PL index for a ploidy.
 Calculation in @see #calculatePLIndexToAlleleIndices</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>altAlleles</code> - Number of alternate alleles</dd>
<dd><code>ploidy</code> - Number of chromosomes in set</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>PL index to the alleles of general ploidy over all possible alternate alleles</dd>
</dl>
</li>
</ul>
<a name="numLikelihoods-int-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>numLikelihoods</h4>
<pre>public static&nbsp;int&nbsp;numLikelihoods(int&nbsp;numAlleles,
                                 int&nbsp;ploidy)</pre>
<div class="block">Compute how many likelihood elements are associated with the given number of alleles
 Equivalent to asking in how many ways N non-negative integers can add up to P is S(N,P)
 where P = ploidy (number of chromosomes) and N = total # of alleles.
 Each chromosome can be in one single state (0,...,N-1) and there are P of them.
 Naive solution would be to store N*P likelihoods, but this is not necessary because we can't distinguish chromosome states, but rather
 only total number of alt allele counts in all chromosomes.

 For example, S(3,2) = 6: For alleles A,B,C, on a diploid organism we have six possible genotypes:
 AA,AB,BB,AC,BC,CC.
 Another way of expressing is with vector (#of A alleles, # of B alleles, # of C alleles)
 which is then, for ordering above, (2,0,0), (1,1,0), (0,2,0), (1,1,0), (0,1,1), (0,0,2)
 In general, for P=2 (regular biallelic), then S(N,2) = N*(N+1)/2

 Note this method caches the value for most common num Allele / ploidy combinations for efficiency

 Recursive implementation:
   S(N,P) = sum_{k=0}^P S(N-1,P-k)
  because if we have N integers, we can condition 1 integer to be = k, and then N-1 integers have to sum to P-K
 With initial conditions
   S(N,1) = N  (only way to have N integers add up to 1 is all-zeros except one element with a one. There are N of these vectors)
   S(1,P) = 1 (only way to have 1 integer add to P is with that integer P itself).</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>numAlleles</code> - Number of alleles (including ref)</dd>
<dd><code>ploidy</code> - Ploidy, or number of chromosomes in set</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>Number of likelihood elements we need to hold.</dd>
</dl>
</li>
</ul>
<a name="calculatePLindex-int-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>calculatePLindex</h4>
<pre>public static&nbsp;int&nbsp;calculatePLindex(int&nbsp;allele1Index,
                                   int&nbsp;allele2Index)</pre>
</li>
</ul>
<a name="getAllelePair-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAllelePair</h4>
<pre>public static&nbsp;<a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.GenotypeLikelihoodsAllelePair.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods.GenotypeLikelihoodsAllelePair</a>&nbsp;getAllelePair(int&nbsp;PLindex)</pre>
<div class="block">Get the diploid allele index pair for the given PL index</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>PLindex</code> - the PL index</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the diploid allele index pair</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.IllegalStateException</code> - if PLindex is negative value or greater than the cache computed by @see #calculateDiploidPLcache</dd>
</dl>
</li>
</ul>
<a name="initializeAnyploidPLIndexToAlleleIndices-int-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>initializeAnyploidPLIndexToAlleleIndices</h4>
<pre>public static&nbsp;void&nbsp;initializeAnyploidPLIndexToAlleleIndices(int&nbsp;altAlleles,
                                                            int&nbsp;ploidy)</pre>
<div class="block">Initialize cache of allele anyploid indices
 If initialized multiple times with the same ploidy, the alternate alleles from the last initialization will be used</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>altAlleles</code> - number of alternate alleles</dd>
<dd><code>ploidy</code> - number of chromosomes</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.IllegalArgumentException</code> - if altAlleles or ploidy &lt= 0</dd>
</dl>
</li>
</ul>
<a name="getAlleles-int-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAlleles</h4>
<pre>public static&nbsp;java.util.List&lt;java.lang.Integer&gt;&nbsp;getAlleles(int&nbsp;PLindex,
                                                           int&nbsp;ploidy)</pre>
<div class="block">Get the allele ploidy indices for the given PL index
 Must use the same ploidy as @see #initializeAnyploidPLIndexToAlleleIndices</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>PLindex</code> - the PL index</dd>
<dd><code>ploidy</code> - number of chromosomes</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the ploidy allele indices</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code>java.lang.IllegalStateException</code> - if @see #anyploidPloidyToPLIndexToAlleleIndices does not contain the requested ploidy or PL index</dd>
</dl>
</li>
</ul>
<a name="getAllelePairUsingDeprecatedOrdering-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAllelePairUsingDeprecatedOrdering</h4>
<pre>@Deprecated
public static&nbsp;<a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.GenotypeLikelihoodsAllelePair.html" title="class in htsjdk.variant.variantcontext">GenotypeLikelihoods.GenotypeLikelihoodsAllelePair</a>&nbsp;getAllelePairUsingDeprecatedOrdering(int&nbsp;PLindex)</pre>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
<div class="block">get the allele index pair for the given PL using the deprecated PL ordering:
    AA,AB,AC,AD,BB,BC,BD,CC,CD,DD instead of AA,AB,BB,AC,BC,CC,AD,BD,CD,DD.
 Although it's painful to keep this conversion around, our DiploidSNPGenotypeLikelihoods class uses the deprecated
    ordering and I know with certainty that external users have built code on top of it; changing it now would
    cause a whole lot of heartache for our collaborators, so for now at least there's a standard conversion method.
 This method assumes at most 3 alternate alleles.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>PLindex</code> - the PL index</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the allele index pair</dd>
</dl>
</li>
</ul>
<a name="getPLIndecesOfAlleles-int-int-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getPLIndecesOfAlleles</h4>
<pre>public static&nbsp;int[]&nbsp;getPLIndecesOfAlleles(int&nbsp;allele1Index,
                                          int&nbsp;allele2Index)</pre>
<div class="block">get the PL indexes (AA, AB, BB) for the given allele pair; assumes allele1Index &lt;= allele2Index.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>allele1Index</code> - the index in VariantContext.getAllele() of the first allele</dd>
<dd><code>allele2Index</code> - the index in VariantContext.getAllele() of the second allele</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the PL indexes</dd>
</dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GenotypeLikelihoods.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-all.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../htsjdk/variant/variantcontext/GenotypeBuilder.html" title="class in htsjdk.variant.variantcontext"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../../htsjdk/variant/variantcontext/GenotypeLikelihoods.GenotypeLikelihoodsAllelePair.html" title="class in htsjdk.variant.variantcontext"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?htsjdk/variant/variantcontext/GenotypeLikelihoods.html" target="_top">Frames</a></li>
<li><a href="GenotypeLikelihoods.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.class.summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
