<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc -->
<title>Uses of Class picard.vcf.GenotypeConcordanceCounts (PICARD JDK API Documentation)</title>
<link rel="stylesheet" type="text/css" href="../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Class picard.vcf.GenotypeConcordanceCounts (PICARD JDK API Documentation)";
        }
    }
    catch(err) {
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../../../overview-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-all.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?picard/vcf/class-use/GenotypeConcordanceCounts.html" target="_top">Frames</a></li>
<li><a href="GenotypeConcordanceCounts.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Class picard.vcf.GenotypeConcordanceCounts" class="title">Uses of Class<br>picard.vcf.GenotypeConcordanceCounts</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#picard.vcf">picard.vcf</a></td>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="picard.vcf">
<!--   -->
</a>
<h3>Uses of <a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a> in <a href="../../../picard/vcf/package-summary.html">picard.vcf</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../picard/vcf/package-summary.html">picard.vcf</a> declared as <a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a></code></td>
<td class="colLast"><span class="typeNameLabel">GenotypeConcordance.</span><code><span class="memberNameLink"><a href="../../../picard/vcf/GenotypeConcordance.html#indelCounter">indelCounter</a></span></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a></code></td>
<td class="colLast"><span class="typeNameLabel">GenotypeConcordance.</span><code><span class="memberNameLink"><a href="../../../picard/vcf/GenotypeConcordance.html#snpCounter">snpCounter</a></span></code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../picard/vcf/package-summary.html">picard.vcf</a> that return <a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a></code></td>
<td class="colLast"><span class="typeNameLabel">GenotypeConcordance.</span><code><span class="memberNameLink"><a href="../../../picard/vcf/GenotypeConcordance.html#getIndelCounter--">getIndelCounter</a></span>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a></code></td>
<td class="colLast"><span class="typeNameLabel">GenotypeConcordance.</span><code><span class="memberNameLink"><a href="../../../picard/vcf/GenotypeConcordance.html#getSnpCounter--">getSnpCounter</a></span>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../picard/vcf/package-summary.html">picard.vcf</a> with parameters of type <a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">GenotypeConcordance.</span><code><span class="memberNameLink"><a href="../../../picard/vcf/GenotypeConcordance.html#addMissingTruthAndMissingCallStates-double-long-picard.vcf.GenotypeConcordanceCounts-">addMissingTruthAndMissingCallStates</a></span>(double&nbsp;numVariants,
                                   long&nbsp;intervalBaseCount,
                                   <a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&nbsp;counter)</code>
<div class="block">Method to add missing sites that are KNOWN to be HOM_REF in the case of the NIST truth data set.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">GenotypeConcordance.</span><code><span class="memberNameLink"><a href="../../../picard/vcf/GenotypeConcordance.html#outputDetailMetricsFile-htsjdk.variant.variantcontext.VariantContext.Type-htsjdk.samtools.metrics.MetricsFile-picard.vcf.GenotypeConcordanceCounts-java.lang.String-java.lang.String-boolean-boolean-">outputDetailMetricsFile</a></span>(htsjdk.variant.variantcontext.VariantContext.Type&nbsp;variantType,
                       htsjdk.samtools.metrics.MetricsFile&lt;<a href="../../../picard/vcf/GenotypeConcordanceDetailMetrics.html" title="class in picard.vcf">GenotypeConcordanceDetailMetrics</a>,?&gt;&nbsp;genotypeConcordanceDetailMetricsFile,
                       <a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&nbsp;counter,
                       java.lang.String&nbsp;truthSampleName,
                       java.lang.String&nbsp;callSampleName,
                       boolean&nbsp;missingSitesHomRef,
                       boolean&nbsp;outputAllRows)</code>
<div class="block">Outputs the detailed statistics tables for SNP and Indel match categories.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../picard/vcf/package-summary.html">picard.vcf</a> with type arguments of type <a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><span class="typeNameLabel">GenotypeConcordance.</span><code><span class="memberNameLink"><a href="../../../picard/vcf/GenotypeConcordance.html#classifyVariants-java.util.Optional-java.lang.String-java.util.Optional-java.lang.String-java.util.Optional-java.util.Optional-int-int-">classifyVariants</a></span>(java.util.Optional&lt;htsjdk.variant.variantcontext.VariantContext&gt;&nbsp;truthContext,
                java.lang.String&nbsp;truthSample,
                java.util.Optional&lt;htsjdk.variant.variantcontext.VariantContext&gt;&nbsp;callContext,
                java.lang.String&nbsp;callSample,
                java.util.Optional&lt;<a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&gt;&nbsp;snpCounter,
                java.util.Optional&lt;<a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&gt;&nbsp;indelCounter,
                int&nbsp;minGq,
                int&nbsp;minDp)</code>
<div class="block">Attempts to determine the concordance state given the truth and all variant context and optionally increments the genotype concordance
 count for the given variant type (SNP or INDEL).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><span class="typeNameLabel">GenotypeConcordance.</span><code><span class="memberNameLink"><a href="../../../picard/vcf/GenotypeConcordance.html#classifyVariants-java.util.Optional-java.lang.String-java.util.Optional-java.lang.String-java.util.Optional-java.util.Optional-int-int-">classifyVariants</a></span>(java.util.Optional&lt;htsjdk.variant.variantcontext.VariantContext&gt;&nbsp;truthContext,
                java.lang.String&nbsp;truthSample,
                java.util.Optional&lt;htsjdk.variant.variantcontext.VariantContext&gt;&nbsp;callContext,
                java.lang.String&nbsp;callSample,
                java.util.Optional&lt;<a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&gt;&nbsp;snpCounter,
                java.util.Optional&lt;<a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&gt;&nbsp;indelCounter,
                int&nbsp;minGq,
                int&nbsp;minDp)</code>
<div class="block">Attempts to determine the concordance state given the truth and all variant context and optionally increments the genotype concordance
 count for the given variant type (SNP or INDEL).</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../picard/vcf/package-summary.html">picard.vcf</a> with parameters of type <a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/vcf/GenotypeConcordanceContingencyMetrics.html#GenotypeConcordanceContingencyMetrics-htsjdk.variant.variantcontext.VariantContext.Type-picard.vcf.GenotypeConcordanceCounts-java.lang.String-java.lang.String-boolean-">GenotypeConcordanceContingencyMetrics</a></span>(htsjdk.variant.variantcontext.VariantContext.Type&nbsp;variantType,
                                     <a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&nbsp;concordanceCounts,
                                     java.lang.String&nbsp;truthSample,
                                     java.lang.String&nbsp;callSample,
                                     boolean&nbsp;missingSitesFlag)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../picard/vcf/GenotypeConcordanceSummaryMetrics.html#GenotypeConcordanceSummaryMetrics-htsjdk.variant.variantcontext.VariantContext.Type-picard.vcf.GenotypeConcordanceCounts-java.lang.String-java.lang.String-boolean-">GenotypeConcordanceSummaryMetrics</a></span>(htsjdk.variant.variantcontext.VariantContext.Type&nbsp;variantType,
                                 <a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">GenotypeConcordanceCounts</a>&nbsp;concordanceCounts,
                                 java.lang.String&nbsp;truthSample,
                                 java.lang.String&nbsp;callSample,
                                 boolean&nbsp;missingSitesFlag)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../picard/vcf/GenotypeConcordanceCounts.html" title="class in picard.vcf">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../../../overview-tree.html">Tree</a></li>
<li><a href="../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../index-all.html">Index</a></li>
<li><a href="../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../index.html?picard/vcf/class-use/GenotypeConcordanceCounts.html" target="_top">Frames</a></li>
<li><a href="GenotypeConcordanceCounts.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
