<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_111) on Thu Dec 08 15:12:57 EST 2016 -->
<title>CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics (picard 2.8.0-SNAPSHOT API)</title>
<meta name="date" content="2016-12-08">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics (picard 2.8.0-SNAPSHOT API)";
        }
    }
    catch(err) {
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.FILTERING_STRINGENCY.html" title="enum in picard.analysis"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/analysis/CollectWgsMetricsFromSampledSites.html" title="class in picard.analysis"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/analysis/CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html" target="_top">Frames</a></li>
<li><a href="CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.classes.inherited.from.class.picard.analysis.MergeableMetricBase">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#methods.inherited.from.class.picard.analysis.CollectWgsMetrics.WgsMetrics">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li>Method</li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">picard.analysis</div>
<h2 title="Class CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics" class="title">Class CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li>java.lang.Object</li>
<li>
<ul class="inheritance">
<li>htsjdk.samtools.metrics.MetricBase</li>
<li>
<ul class="inheritance">
<li><a href="../../picard/analysis/MergeableMetricBase.html" title="class in picard.analysis">picard.analysis.MergeableMetricBase</a></li>
<li>
<ul class="inheritance">
<li><a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html" title="class in picard.analysis">picard.analysis.CollectWgsMetrics.WgsMetrics</a></li>
<li>
<ul class="inheritance">
<li>picard.analysis.CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>Enclosing class:</dt>
<dd><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.html" title="class in picard.analysis">CollectWgsMetricsFromQuerySorted</a></dd>
</dl>
<hr>
<br>
<pre>public static class <span class="typeNameLabel">CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics</span>
extends <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html" title="class in picard.analysis">CollectWgsMetrics.WgsMetrics</a></pre>
<div class="block">Metrics for evaluating the performance of whole genome sequencing experiments.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="nested.classes.inherited.from.class.picard.analysis.MergeableMetricBase">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;picard.analysis.<a href="../../picard/analysis/MergeableMetricBase.html" title="class in picard.analysis">MergeableMetricBase</a></h3>
<code><a href="../../picard/analysis/MergeableMetricBase.MergeByAdding.html" title="annotation in picard.analysis">MergeableMetricBase.MergeByAdding</a>, <a href="../../picard/analysis/MergeableMetricBase.MergeByAssertEquals.html" title="annotation in picard.analysis">MergeableMetricBase.MergeByAssertEquals</a>, <a href="../../picard/analysis/MergeableMetricBase.MergingIsManual.html" title="annotation in picard.analysis">MergeableMetricBase.MergingIsManual</a>, <a href="../../picard/analysis/MergeableMetricBase.NoMergingIsDerived.html" title="annotation in picard.analysis">MergeableMetricBase.NoMergingIsDerived</a>, <a href="../../picard/analysis/MergeableMetricBase.NoMergingKeepsValue.html" title="annotation in picard.analysis">MergeableMetricBase.NoMergingKeepsValue</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html#MEAN_INSERT_SIZE">MEAN_INSERT_SIZE</a></span></code>
<div class="block">The mean insert size, after filters are applied.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html#PF_ADAPTER_READS">PF_ADAPTER_READS</a></span></code>
<div class="block">The number of PF reads that are unaligned and match to a known adapter sequence right from the
 start of the read.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html#PF_BASES">PF_BASES</a></span></code>
<div class="block">The total number of bases, before any filters are applied.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html#PF_CHIMERIC_PAIRS">PF_CHIMERIC_PAIRS</a></span></code>
<div class="block">The number of PF reads that map outside of a maximum insert size (100kb) or that have
 the two ends mapping to different chromosomes.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html#PF_DUPE_PAIRS">PF_DUPE_PAIRS</a></span></code>
<div class="block">The number of duplicate read pairs, before any filters are applied.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html#PF_NOISE_READS">PF_NOISE_READS</a></span></code>
<div class="block">The number of PF reads that are marked as noise reads.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html#PF_ORIENTED_PAIRS">PF_ORIENTED_PAIRS</a></span></code>
<div class="block">The number of read pairs with standard orientations from which to calculate mean insert size, after filters are applied.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html#PF_PASSING_BASES">PF_PASSING_BASES</a></span></code>
<div class="block">The number of passing bases, after all filters are applied.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html#PF_READ_PAIRS">PF_READ_PAIRS</a></span></code>
<div class="block">The number of read pairs, before all filters are applied.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html#PF_READS_ALIGNED">PF_READS_ALIGNED</a></span></code>
<div class="block">The number of aligned reads, before any filters are applied.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.FILTERING_STRINGENCY.html" title="enum in picard.analysis">CollectWgsMetricsFromQuerySorted.FILTERING_STRINGENCY</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html#TYPE">TYPE</a></span></code>
<div class="block">Identifier for metrics type</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="fields.inherited.from.class.picard.analysis.CollectWgsMetrics.WgsMetrics">
<!--   -->
</a>
<h3>Fields inherited from class&nbsp;picard.analysis.<a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html" title="class in picard.analysis">CollectWgsMetrics.WgsMetrics</a></h3>
<code><a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#coverageCap">coverageCap</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#GENOME_TERRITORY">GENOME_TERRITORY</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#HET_SNP_Q">HET_SNP_Q</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#HET_SNP_SENSITIVITY">HET_SNP_SENSITIVITY</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#highQualityDepthHistogram">highQualityDepthHistogram</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#intervals">intervals</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#MAD_COVERAGE">MAD_COVERAGE</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#MEAN_COVERAGE">MEAN_COVERAGE</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#MEDIAN_COVERAGE">MEDIAN_COVERAGE</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_100X">PCT_100X</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_10X">PCT_10X</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_15X">PCT_15X</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_1X">PCT_1X</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_20X">PCT_20X</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_25X">PCT_25X</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_30X">PCT_30X</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_40X">PCT_40X</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_50X">PCT_50X</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_5X">PCT_5X</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_60X">PCT_60X</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_70X">PCT_70X</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_80X">PCT_80X</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_90X">PCT_90X</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_EXC_BASEQ">PCT_EXC_BASEQ</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_EXC_CAPPED">PCT_EXC_CAPPED</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_EXC_DUPE">PCT_EXC_DUPE</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_EXC_MAPQ">PCT_EXC_MAPQ</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_EXC_OVERLAP">PCT_EXC_OVERLAP</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_EXC_TOTAL">PCT_EXC_TOTAL</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#PCT_EXC_UNPAIRED">PCT_EXC_UNPAIRED</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#SD_COVERAGE">SD_COVERAGE</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#theoreticalHetSensitivitySampleSize">theoreticalHetSensitivitySampleSize</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#unfilteredBaseQHistogram">unfilteredBaseQHistogram</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#unfilteredDepthHistogram">unfilteredDepthHistogram</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html#QuerySortedSeqMetrics--">QuerySortedSeqMetrics</a></span>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.picard.analysis.CollectWgsMetrics.WgsMetrics">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;picard.analysis.<a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html" title="class in picard.analysis">CollectWgsMetrics.WgsMetrics</a></h3>
<code><a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#calculateDerivedFields--">calculateDerivedFields</a>, <a href="../../picard/analysis/CollectWgsMetrics.WgsMetrics.html#merge-picard.analysis.MergeableMetricBase-">merge</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.picard.analysis.MergeableMetricBase">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;picard.analysis.<a href="../../picard/analysis/MergeableMetricBase.html" title="class in picard.analysis">MergeableMetricBase</a></h3>
<code><a href="../../picard/analysis/MergeableMetricBase.html#canMerge-picard.analysis.MergeableMetricBase-">canMerge</a>, <a href="../../picard/analysis/MergeableMetricBase.html#merge-java.util.Collection-">merge</a>, <a href="../../picard/analysis/MergeableMetricBase.html#mergeIfCan-picard.analysis.MergeableMetricBase-">mergeIfCan</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.htsjdk.samtools.metrics.MetricBase">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;htsjdk.samtools.metrics.MetricBase</h3>
<code>equals, hashCode, toString</code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.Object</h3>
<code>clone, finalize, getClass, notify, notifyAll, wait, wait, wait</code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="TYPE">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>TYPE</h4>
<pre>public&nbsp;<a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.FILTERING_STRINGENCY.html" title="enum in picard.analysis">CollectWgsMetricsFromQuerySorted.FILTERING_STRINGENCY</a> TYPE</pre>
<div class="block">Identifier for metrics type</div>
</li>
</ul>
<a name="PF_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PF_BASES</h4>
<pre>public&nbsp;long PF_BASES</pre>
<div class="block">The total number of bases, before any filters are applied.</div>
</li>
</ul>
<a name="PF_PASSING_BASES">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PF_PASSING_BASES</h4>
<pre>public&nbsp;long PF_PASSING_BASES</pre>
<div class="block">The number of passing bases, after all filters are applied.</div>
</li>
</ul>
<a name="PF_READ_PAIRS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PF_READ_PAIRS</h4>
<pre>public&nbsp;long PF_READ_PAIRS</pre>
<div class="block">The number of read pairs, before all filters are applied.</div>
</li>
</ul>
<a name="PF_DUPE_PAIRS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PF_DUPE_PAIRS</h4>
<pre>public&nbsp;long PF_DUPE_PAIRS</pre>
<div class="block">The number of duplicate read pairs, before any filters are applied.</div>
</li>
</ul>
<a name="PF_READS_ALIGNED">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PF_READS_ALIGNED</h4>
<pre>public&nbsp;long PF_READS_ALIGNED</pre>
<div class="block">The number of aligned reads, before any filters are applied.</div>
</li>
</ul>
<a name="PF_NOISE_READS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PF_NOISE_READS</h4>
<pre>public&nbsp;long PF_NOISE_READS</pre>
<div class="block">The number of PF reads that are marked as noise reads.  A noise read is one which is composed
 entirely of A bases and/or N bases. These reads are marked as they are usually artifactual and
 are of no use in downstream analysis.</div>
</li>
</ul>
<a name="PF_CHIMERIC_PAIRS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PF_CHIMERIC_PAIRS</h4>
<pre>public&nbsp;long PF_CHIMERIC_PAIRS</pre>
<div class="block">The number of PF reads that map outside of a maximum insert size (100kb) or that have
 the two ends mapping to different chromosomes.</div>
</li>
</ul>
<a name="PF_ADAPTER_READS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PF_ADAPTER_READS</h4>
<pre>public&nbsp;long PF_ADAPTER_READS</pre>
<div class="block">The number of PF reads that are unaligned and match to a known adapter sequence right from the
 start of the read.</div>
</li>
</ul>
<a name="PF_ORIENTED_PAIRS">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>PF_ORIENTED_PAIRS</h4>
<pre>public&nbsp;long PF_ORIENTED_PAIRS</pre>
<div class="block">The number of read pairs with standard orientations from which to calculate mean insert size, after filters are applied.</div>
</li>
</ul>
<a name="MEAN_INSERT_SIZE">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>MEAN_INSERT_SIZE</h4>
<pre>public&nbsp;double MEAN_INSERT_SIZE</pre>
<div class="block">The mean insert size, after filters are applied.</div>
</li>
</ul>
</li>
</ul>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="QuerySortedSeqMetrics--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>QuerySortedSeqMetrics</h4>
<pre>public&nbsp;QuerySortedSeqMetrics()</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../index-all.html">Index</a></li>
<li><a href="../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../picard/analysis/CollectWgsMetricsFromQuerySorted.FILTERING_STRINGENCY.html" title="enum in picard.analysis"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../picard/analysis/CollectWgsMetricsFromSampledSites.html" title="class in picard.analysis"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../index.html?picard/analysis/CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html" target="_top">Frames</a></li>
<li><a href="CollectWgsMetricsFromQuerySorted.QuerySortedSeqMetrics.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.classes.inherited.from.class.picard.analysis.MergeableMetricBase">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#methods.inherited.from.class.picard.analysis.CollectWgsMetrics.WgsMetrics">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li>Method</li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</body>
</html>
